### DISCUSSION TITLE
[microsoft/MixedRealityToolkit-Unity] MRTK3 tutorial - Integrate Azure Custom Vision

### DISCUSSION THREAD

**Discussion by dldnwlsgd** [MixedRealityToolkit/MixedRealityToolkit-Unity#537] [Q&A]
In the process of integrating custom vision, I created a custom vision account and entered the key and ID into the 'ObjectDetectionManager'. Afterwards, when I press the Train button in Unity Play mode, the following error appears:

I confirmed that the photo was uploaded to the Custom Vision website, but since then, when I press the "Train" button in Unity, an error continues to appear. Can anyone help me with what problem I'm having?

`Exception: PermissionDenied
MRTK.Tutorials.AzureCloudServices.Scripts.Managers.ObjectDetectionManager.TrainProject () (at Assets/MRTK.Tutorials.AzureCloudServices/Scripts/Managers/ObjectDetectionManager.cs:113)
MRTK.Tutorials.AzureCloudServices.Scripts.Controller.ComputerVisionController.StartModelTraining () (at Assets/MRTK.Tutorials.AzureCloudServices/Scripts/Controller/ComputerVisionController.cs:199)
System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) (at <9d9536d9127f4a489d989c7a566aee1c>:0)
UnityEngine.UnitySynchronizationContext+WorkRequest.Invoke () (at <120b8d04add741329ccd415c000fb666>:0)
UnityEngine.UnitySynchronizationContext.Exec () (at <120b8d04add741329ccd415c000fb666>:0)
UnityEngine.UnitySynchronizationContext.ExecuteTasks () (at <120b8d04add741329ccd415c000fb666>:0)`

**Comment by AMollis** [comment]
@marlenaklein-msft should be able to assist here

**Reply by AMollis** [reply]
Also @dldnwlsgd , these learning paths are part of setting up a HoloLens project, and are maintained by Microsoft.   If there is a bug here, please open it in our "Learning Path" repository at https://github.com/microsoft/MixedRealityLearning/issues

**Comment by marlenaklein-msft** [comment]
My bad @AMollis, I thought the bug would be logged here but yes it should be MixedRealityLearning. 
	
@dldnwlsgd to make sure it's not an authentication problem, can you please confirm that you have one key/endpoint pair one for your custom vision resource and another set of credentials for the corresponding prediction resource?  (The [tutorial](https://learn.microsoft.com/en-us/training/modules/azure-cloud-services-tutorials/7-exercise-integrate-azure-custom-vision) will describe where to set these credentials. If you have one but not the other, you may be able to authenticate only some of the steps).
	
If the credentials are set properly, there may be a problem with the training itself. To train a model, you need to have at least **two tags and five images per tag**. The tutorial links instructions for how to add a tag and five images from the custom vision dashboard, which is a necessary step for the later training to work from HL2. Another thing is to make sure the tag used for those first five images added from the dashboard is different from the tag you are using when uploading the second batch of images from HL2, so you hit that two tag requirement. 
	
<img width="201" alt="Screenshot 2023-11-03 114709" src="https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/125413876/87b71bc5-ae3e-441a-94ea-8145eab69115">
	
On the [dashboard](https://www.customvision.ai/projects), can you make sure that you have at least two tags and five images for every tag? If there are and you are still seeing that error, then try training the model manually from the dashboard. This will confirm if it's a problem with the request.
	
<img width="833" alt="Screenshot 2023-11-03 112359" src="https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/125413876/a350b59e-25a8-4139-8b68-69ef73150795">
	
If you are still encountering any problems, could you please file an issue on MixedRealityLearning? (sorry for the misdirection earlier)
